Source/Fuse.GeoLocation/Android/AndroidLocationProvider.uno

```diff
@@ -173,17 +209,44 @@ namespace Fuse.GeoLocation
 		
 		public void StartListening(Action<Location> onLocationChanged, Action<Uno.Exception> onLocationError, int minimumReportInterval, double desiredAccuracyInMeters)
 		{
-			if (_locationManager != null && !_started)
-			{
-				_onLocationChanged = onLocationChanged;
+			if (!IsLocationEnabled()) {
+
+				var permissions = new PlatformPermission[] 
+				{
+					Permissions.Android.INTERNET,
+					Permissions.Android.ACCESS_COARSE_LOCATION,
+					Permissions.Android.ACCESS_FINE_LOCATION
+				};
 				
-				if(IsNetworkEnabled(_locationManager))
-					RequestNetworkLocationUpdates(_locationManager, minimumReportInterval, desiredAccuracyInMeters, _updateListener);
+				Permissions.Request(permissions).Then(OnPermissionsResult, OnPermissionsError);
```

Comment:
Why is this needed again? It's already done in Init?

Reply:
Its because the user can disable geo-location outside of the app, after initially saying yes.

Reply:
Right, makes sense. But perhaps this should be a helper-method, so we don't have to repeat ourselves, then?

Reply:
There is a helper method: IsLocationEnabled()

Which is being used here to determine if it is enabled, you gotta do it on every "GPS call" as the user can turn off at anytime.

Reply:
That's a different helper than what I'm proposing. I'm talking about these lines:
```java
var permissions = new PlatformPermission[] 
{
    Permissions.Android.INTERNET,
    Permissions.Android.ACCESS_COARSE_LOCATION,
    Permissions.Android.ACCESS_FINE_LOCATION
};
Permissions.Request(permissions).Then(OnPermissionsResult, OnPermissionsError);
```
...as these are exactly the same, and if we need to change the list of permissions we now have to update two locations.

Reply:
oh I see, my bad, I actually can't think how to make that dynamic right now, ideas? 😄

Reply:
Add the following helper method.

```
		void RequestPermissions()
		{
			var permissions = new PlatformPermission[] 
			{
				Permissions.Android.INTERNET,
				Permissions.Android.ACCESS_COARSE_LOCATION,
				Permissions.Android.ACCESS_FINE_LOCATION
			};
			
			Permissions.Request(permissions).Then(OnPermissionsResult, OnPermissionsError);
		}
```

... then replace the two identical permission-snippets in this class with calls to your new `RequestPermissions()` helper. :)

Reply:
Oh I see, I thought @kusma was talking about having those permissions editable in like a .uxl or summin, will make the change soon...

Reply:
Yup, that's exactly what I meant! Sorry if I was unclear :)